src/azure-cli/azure/cli/command_modules/dms/custom.py

```diff
@@ -185,19 +205,242 @@ def list_tasks(client, resource_group_name, service_name, project_name, task_typ
+        GetInput = get_migrate_mysql_to_azuredbformysql_sync_input
+    elif st == ScenarioType.postgres_azurepostgres_online:
+        TaskProperties = MigratePostgreSqlAzureDbForPostgreSqlSyncTaskProperties
+        GetInput = get_migrate_postgresql_to_azuredbforpostgresql_sync_input
+    else:
+        raise CLIError("The supplied source, target, and task type is not supported for migration.")
+
+    return get_task_properties(st,
+                               GetInput,
+                               TaskProperties,
+                               database_options_json,
+                               source_connection_info,
+                               target_connection_info,
+                               enable_schema_validation,
+                               enable_data_integrity_validation,
+                               enable_query_analysis_validation)
+
+
+def get_task_properties(scenario_type,
+                        input_func,
+                        task_properties_type,
+                        options_json,
+                        source_connection_info,
+                        target_connection_info,
+                        enable_schema_validation,
+                        enable_data_integrity_validation,
+                        enable_query_analysis_validation):
+    if source_connection_info is None and target_connection_info is None:
+        task_input = input_func(options_json)
+    elif scenario_type == ScenarioType.sql_sqldb_offline:
+        task_input = input_func(
+            options_json,
+            source_connection_info,
+            target_connection_info,
+            enable_schema_validation,
+            enable_data_integrity_validation,
+            enable_query_analysis_validation)
+    else:
+        task_input = input_func(
+            options_json,
+            source_connection_info,
+            target_connection_info)
+
+    task_properties_params = {'input': task_input}
+
+    return task_properties_type(**task_properties_params)
+
+
+def get_scenario_type(source_platform, target_platform, task_type=""):
+    if source_platform == "sql" and target_platform == "sqldb":
+        scenario_type = ScenarioType.sql_sqldb_offline if not task_type or "offline" in task_type else \
+            ScenarioType.unknown
+    elif source_platform == "mysql" and target_platform == "azuredbformysql":
+        scenario_type = ScenarioType.mysql_azuremysql_online if not task_type or "online" in task_type else \
+            ScenarioType.unknown
+    elif source_platform == "postgresql" and target_platform == "azuredbforpostgresql":
+        scenario_type = ScenarioType.postgres_azurepostgres_online if not task_type or "online" in task_type else \
+            ScenarioType.unknown
+    else:
+        scenario_type = ScenarioType.unknown
+
+    return scenario_type
+
+
+class ScenarioType(Enum):
 
-def create_sql_connection_info(connection_info_json, prompt_prefix):
```

Comment:
why removing these code?

Reply:
This was refactored into custom.create_connection.